https://bugs.gentoo.org/show_bug.cgi?id=281968
https://bugs.python.org/issue6731

--- setup.py
+++ setup.py
@@ -56,6 +56,8 @@
 else:
     disable_ssl = 0
 
+exit_status = 0
+
 def add_dir_to_list(dirlist, dir):
     """Add the directory 'dir' to the list 'dirlist' (after any relative
     directories) if:
@@ -297,7 +299,10 @@
                   " detect_modules() for the module's name.")
             print()
 
+        global exit_status
+
         if self.failed:
+            exit_status = 1
             failed = self.failed[:]
             print()
             print("Failed to build these modules:")
@@ -305,6 +310,7 @@
             print()
 
         if self.failed_on_import:
+            exit_status = 1
             failed = self.failed_on_import[:]
             print()
             print("Following modules built successfully"
@@ -2251,6 +2257,7 @@
           scripts = ["Tools/scripts/pydoc3", "Tools/scripts/idle3",
                      "Tools/scripts/2to3", "Tools/scripts/pyvenv"]
         )
+    sys.exit(exit_status)
 
 # --install-platlib
 if __name__ == '__main__':
